modules/govuk/manifests/apps/ckan.pp

```diff
@@ -173,5 +184,17 @@
       owner  => 'deploy',
       group  => 'deploy',
     }
+
+    exec { 'symlink_pycsw_config':
+      command => "test -e ${ckan_home}/pycsw.cfg || ln -s ${pycsw_src}/default.cfg ${ckan_home}/pycsw.cfg",
+    }
+
+    exec { 'setup_pycsw_tables':
+      command => "test -e ${ckan_home}/pycsw_tables_created.tmp || ${ckan_bin}/paster --plugin=ckanext-spatial ckan-pycsw setup -p ${ckan_home}/pycsw.cfg && sudo touch ${ckan_home}/pycsw_tables_created.tmp",
+    }
+
+    exec { 'startup_pycsw_server':
+      command => "ps -C \"python */pycsw/csw.wsgi\" >/dev/null && echo \"pycsw is running\" || eval \". ${ckan_bin}/activate && python ${pycsw_src}/csw.wsgi &\"",
+    }
```

Comment:
Have you considered using an Upstart service for this?

Reply:
No, but I'll have a look into it

Reply:
Looks like an upstart service is now being used.